### ISSUE TITLE
Use the Vite error overlay for more errors at dev time

### ISSUE THREAD

**Issue by geoffrich** [sveltejs/kit#9704]
### Describe the problem

During development, some runtime errors only show a message in the console. For example, if you cause an error in a `load` function or during rendering of a component, SvelteKit's error page is displayed and you need to look at the dev console to figure out where the error is. New users may not realize that they need to look at the console for this.

It would be nice if the error and stack trace was shown on the page as well.

This was called out as accessibility feedback in https://github.com/sveltejs/svelte/issues/8508#issuecomment-1512662903, though I think surfacing errors better is more broadly applicable.

### Describe the proposed solution

Expose runtime errors via Vite's error overlay, similar to how compilation errors are displayed.

### Alternatives considered

Do nothing and expect people to keep looking at the dev console.

### Importance

nice to have

### Additional Information

I'm unclear whether this depends on https://github.com/vitejs/vite/issues/2076 or not - parts of this may be blocked by Vite.

**Comment by dummdidumm** [comment]
On tricky part is that errors on the client also need to get the error overlay treatment. Either we send a web socket message to the server to send one back to open it on the client, or we somehow hack our way into Vite's `createErrorOverlay` method.

**Comment by BYK** [comment]
If anyone is still interested, we did something similar to @dummdidumm's suggestion here: https://spotlightjs.com/setup/vite
